### ISSUE TITLE
Freezing,  Refreshing of Menu Items and More

### ISSUE THREAD

**Issue by raythw** [matryer/xbar#780]
It looks like some great work done here, but there are some problems:

1) On installing plug-ins xbar often gets stuck (the box at the top has a spinner in it) and I have to quit it and restart.

2) Plug-in menu items that should refresh with a change in situation, such External IP country flag with the kind of internet connection, don't always do so, especially if the computer has been sleeping in between times. Once a menu item gets stuck in the wrong state you have to quit xbar.

3) Menu items can be Command-dragged to go in between other menu items, but always return to the left when refreshed. I can understand this may be impossible to change.

4) If someone could create a plug-in that worked to show the state of ProtonVPN it would be much appreciated!

Many thanks,

Ray

**Comment by andrewferrier** [comment]
I am having the same issue as @raythw describes above in (2).

**Comment by mmcnl** [comment]
I'm also seeing the case in (2) above, where the items get stuck and won't update. Refreshing the plugin doesn't help and the only solution I've found is restarting xbar. This seems to happen after awakening laptop from sleep.

**Comment by effektsvk** [comment]
Yeah, I'm seeing the second case as well. But I noticed that it's actually happening when I'm switching network connections too often (for example turning off Wi-Fi and switching to USB hotspot from iOS device). My scripts are running every 15 seconds.
